File: code/jvm/http-api/src/main/java/pt/isel/api/users/RegisterInput.java
package pt.isel.api.users;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record RegisterInput(
        @NotBlank @Size(min = 1, max = 30) String username,
        @NotBlank String password,
        String invitationToken
) {
}

File: code/jvm/http-api/src/main/kotlin/pt/isel/api/users/RegisterInput.kt
package pt.isel.api.users

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class RegisterInput(
    @field:NotBlank
    @field:Size(min = 1, max = 30)
    val username: String,
    @field:NotBlank
    val password: String,
    val invitationToken: String? = null,
)
